solr/core/src/java/org/apache/solr/update/SolrIndexConfig.java

```diff
@@ -285,6 +294,21 @@ public IndexWriterConfig toIndexWriterConfig(SolrCore core) throws IOException {
       iwc.setMergedSegmentWarmer(warmer);
     }
 
+    if (segmentSort != null) {
+      try {
+        SegmentSort sortEnum = SegmentSort.valueOf(segmentSort);
+        LeafSorter sorter = new SegmentTimeLeafSorter(sortEnum);
+        Comparator<LeafReader> leafSorter = sorter.getLeafSorter();
+        if (leafSorter != null) {
+          iwc.setLeafSorter(leafSorter);
+          if (log.isDebugEnabled()) {
+            log.debug("Segment sort enabled:  {}", sorter.toString());
```

Comment:
no toString() needed... then you could drop the isDebugEnabled as well (AFAIK).

Reply:
remove toString()